zerver/tests/test_webhooks_common.py

```diff
@@ -117,3 +120,67 @@ def test_missing_event_header(self) -> None:
 
     def get_body(self, fixture_name: str) -> str:
         return self.webhook_fixture_data("groove", fixture_name, file_type="json")
+
+class TestHttpApiCallback(ZulipTestCase):
+
+    def test_http_api_callback_with_unsupported_auth_strategy(self) -> None:
+        url = "https://example.com/api/v2/endpoint"
+        webhook_bot = get_user("webhook-bot@zulip.com", get_realm("zulip"))
+        with self.assertRaises(ValueError):
+            http_api_callback(webhook_bot, 9999, url)
+
+    def test_http_api_callback_with_simple_token_auth_strategy_and_missing_config(self) -> None:
+        url = "https://example.com/api/v2/endpoint"
+        webhook_bot = get_user("webhook-bot@zulip.com", get_realm("zulip"))
+
+        response = http_api_callback(webhook_bot, AuthStrategies.SIMPLE_TOKEN, url)
+        self.assertIsNone(response)
+
+        BotConfigData.objects.create(bot_profile=webhook_bot,
+                                     key="callback_auth_param",
+                                     value="token")
+        response = http_api_callback(webhook_bot, AuthStrategies.SIMPLE_TOKEN, url)
+        self.assertIsNone(response)
+
+    @patch('requests.get')
+    def test_http_api_callback_with_simple_token_auth_strategy_for_success(
+            self, requests_get_mock: MagicMock) -> None:
+        url = "https://example.com/api/v2/endpoint"
+        msg = {"message": "Successful!"}
+        webhook_bot = get_user("webhook-bot@zulip.com", get_realm("zulip"))
+
+        requests_get_mock.return_value = SimpleNamespace(status_code=200, text=json.dumps(msg))
```

Comment:
You could also consider using `httpretty` for your HTTP request mocking, similar to what we do in `test_auth_backends.py`.  I'm not sure if it's actually cleaner; just wanted to mention it.